src/misc/lv_utils.c

```diff
@@ -57,6 +58,35 @@ void * _lv_utils_bsearch(const void * key, const void * base, uint32_t n, uint32
     return NULL;
 }
 
+lv_result_t lv_draw_buf_save_file(const lv_draw_buf_t * draw_buf, const char * path)
+{
+    lv_fs_file_t file;
+    lv_fs_res_t res = lv_fs_open(&file, path, LV_FS_MODE_WR);
+    if(res != LV_FS_RES_OK) {
+        LV_LOG_ERROR("create file %s failed", path);
+        return LV_RESULT_INVALID;
+    }
+
+    uint32_t bw;
+    res = lv_fs_write(&file, &draw_buf->header, sizeof(draw_buf->header), &bw);
+    if(res != LV_FS_RES_OK || bw != sizeof(draw_buf->header)) {
+        LV_LOG_ERROR("write draw_buf->header failed");
+        lv_fs_close(&file);
+        return LV_RESULT_INVALID;
+    }
+
+    res = lv_fs_write(&file, draw_buf->data, draw_buf->data_size, &bw);
```

Comment:
The draw_buf->data may also be changed by GPU, the moment CPU wants to read it, should we invalidate D-cache here too?

Reply:
The quickiest solution would be to duplicate draw buf.
Or to “lock” draw buf, that nobody changes it.
Btw thanks for moving this function out of draw buf core functionality.

Reply:
Duplicate the draw buffer is a CPU action, which still needs to consider cache issue. I'm not sure if we have a central place to take care of cache synchronization and we are free to use it by CPU here if so.

Reply:
I decides to ignore cache issue here, and leave to user code to take care of it since only up level code know if there's cache inconsistency.